lib/rubocop/config_validator.rb

```diff
@@ -271,21 +270,30 @@ def check_cop_config_value(hash, parent = nil)
           supposed_values = 'a boolean'
         elsif key == 'AutoCorrect' && !CONFIG_CHECK_AUTOCORRECTS.include?(value)
           supposed_values = '`always`, `contextual`, `disabled`, or a boolean'
+        elsif key == 'References'
+          warn string_reference_warning(parent)
+          next
```

Comment:
@Earlopain you had concerns about a warning in https://github.com/rubocop/rubocop/pull/13884#discussion_r1965063460

As `References` is a new key, and we disallow unknown keys, it should be safe to assume this key will only be present if someone is migrating from `Reference` (optional), in which case this is the right time to tell them to make it an `Array`. **This should not cause any warnings on any existing configs.**

Given the above,
- are you okay with this warning?
- how would you feel about making it an error rather than just a warning?

Reply:
I'm 100% OK with strictly validating the shape of this new key. If there's logic to warn, it can just as well be logic for raising.

Reply:
Sounds good! I've rebased and changed the validation to raise an error.